tests/test_parsers.py

```diff
@@ -182,6 +182,39 @@ def test_real_time_payment_credit(self):
         self.assertEqual(transaction.amount, 4346722)
         self.assertEqual(len(transaction.rows), 16)
 
+    def test_unsupported_type_code_should_raise_unsupported_exception(self):
```

Comment:
this test was moved from `Bai2FileParserTestCase` to `TransactionDetailParserTestCase` because it was testing only a transaction fragment rather than an entire file